core/site.go

```diff
@@ -82,8 +86,30 @@ func NewSiteFromConfig(
 	site.tariffs = tariffs
 	site.savings = NewSavings(tariffs)
 
+	// maxCurrent is set and > 0: expect load management to be applied
+	if site.MaxCurrent > 0.0 {
+		site.lmEnabled = true
+	} else {
+		// no support for negative values
+		site.lmEnabled = false
+		return nil, errors.New("maxCurrent for the site must not be negative")
+	}
+
 	if site.Meters.GridMeterRef != "" {
 		site.gridMeter = cp.Meter(site.Meters.GridMeterRef)
+		// check we have currents from grid meter. If not, we cannot use load management
+		if site.lmEnabled {
+			if _, ok := site.gridMeter.(api.MeterCurrent); !ok {
+				site.lmEnabled = false
+				return nil, errors.New("can't appy load managament without phase currents from grid meter")
```

Comment:
typo

Brauchts die Variable? Bei dem Fehler sollte aufgrund Fehlkonfiguration abgebrochen werden?

Reply:
s.o., explizite Konfiguration und Lesbarkeit vs implizite Konfig durch Reservierung eines Wertes. `0` halte ich für nicht geeignet, dann ggf `-1`? Soll der Konfig Parameter dann Pflicht werden oder optional?

Reply:
s.o. Ein `maxCurrent==0` kann m.E. nur Lastmanagement inaktiv sein, es sei denn man will den Fall unterstützen einen Strang komplett abzuschalten. Bei statischer Konfiguration ergibt das keinen Sinn.

Reply:
Für die Site: wenn mit `maxCurrent==0` gearbeitet werden soll, muss `maxCurrent` auf type `int` geändert werden, damit `if maxCurrent {..}` funktioniert (alternativ bleibts `float64`, dann aber immer `if maxCurrent > 0.0 {..}`). Alle Ströme werden aber als `float64` momentan abgebildet. Später müsste dann immer explizit auf `float64(maxCurrent)` gecastet werden, wenn mit `maxCurrent` gearbeitet wird.
Vergleiche mit `float64` schlagen in golang gerne fehl aufgrund der internen Präzision, bei `==`.

Bzgl des Abbruchs: ja, ist besser als das LM implizit abzuschalten.